### ISSUE TITLE
[ic-multi-select] bring multi select back in sync with select

### ISSUE THREAD

**Issue by GCHQ-Developer-530** [mi6/ic-ui-kit#2095]
## Summary 
Work has just been completed (https://github.com/mi6/ic-ui-kit/pull/2071) to add an icon to select using the left-icon slot for input component container. This work should be done for multi select as well.

Work also needs to be done to bring multi select back in sync with select. There's been a few select changes over the last few months that aren't in the multi select version of the file.

## 💬 Description 
Multi select should have an icon slot so that users can display an icon before the selected options, like in text field and select. 

This was left out of the original select work because there was work done to the mutation observer to make it so that the slot will show even if it isn't there to begin with (setting it from false to true in the playground story). However the mutation observer seems to be some of the work that hasn't been moved across to multi select from the select updates. So a sync needs to be done first.

## 💰 Use value 
Consistency across components.

**Comment by ad9242** [comment]
this will not be required if #2108 is completed

**Comment by MI6-255** [comment]
On hold, don't refine

**Comment by ad9242** [comment]
closed as we will do #2108 instead
